### PR TITLE
Add Panic Module

### PR DESCRIPTION
**Pull request by SleepingPig35**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new "Panic" module that, when activated, disables all other active modules.
  - Added a command ("panic") that users can execute to disable all enabled modules instantly.

### CODE REVIEW

File: src/main/java/keystrokesmod/utility/Commands.java [hunk 1]
```diff
@@ -266,7 +266,18 @@ public static void rCMD(@NotNull String c) {
...
+                for (Module m : Raven.getModuleManager().getModules()) {
+                    if (m.isEnabled()) {
+                        modulesToDisable.add(m);
+                    }
+                }
+                for (Module m : modulesToDisable) {
+                    m.disable();
+
+                }
+            }else if (args.get(0).equals("rename")) {
```
[the remark this answers is not in the document]
Comment (SleepingPig35) [reply src/main/java/keystrokesmod/utility/Commands.java:280]: What is the point of this? The whole point of the Panic module is to make it look like that there is no Raven bS hack client.

**Comment by SleepingPig35** [comment]
@xia-mc

File: src/main/java/keystrokesmod/module/impl/other/Panic.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        super("Panic", category.other);
+        this.registerSetting(new DescriptionSetting("Disables all modules."));
+    }
+
+    @Override
+    public void onEnable() {
+        List<Module> modulesToDisable = new ArrayList<>();
+        for (Module m : Raven.getModuleManager().getModules()) {
+            if (m.isEnabled()) {
+                modulesToDisable.add(m);
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/module/impl/other/Panic.java:20]: add to arraylist, then disable it? i think it's same as instant disable.
by the way, Module handle the invalid disable, so you needn't check if it's enabled.

File: src/main/java/keystrokesmod/utility/Commands.java [hunk 3]
```diff
@@ -266,7 +266,18 @@ public static void rCMD(@NotNull String c) {
                         print("&a" + module.getName() + " is now visible in HUD", 1);
                     }
                 }
-            } else if (args.get(0).equals("rename")) {
+            } else if (args.get(0).equals("panic")) {
+                List<Module> modulesToDisable = new ArrayList<>();
```
Comment (xia-mc) [inline src/main/java/keystrokesmod/utility/Commands.java:270]: you can do `ModuleManager.panic.enabled()` xd

**Review by xia-mc — APPROVED** [review]
Good.

File: src/main/java/keystrokesmod/module/impl/other/Panic.java (hunk 2 above)
Comment (SleepingPig35) [reply src/main/java/keystrokesmod/module/impl/other/Panic.java:20]: I tried to do that and got a ```java.util.ConcurrentModificationException``` error and this seemed to fix it lol

File: src/main/java/keystrokesmod/utility/Commands.java (hunk 3 above)
Comment (SleepingPig35) [reply src/main/java/keystrokesmod/utility/Commands.java:270]: True

**Comment by TejasLamba2006** [comment]
Just asking but did you add the panic command in the help command?

File: src/main/java/keystrokesmod/module/impl/other/Panic.java (hunk 2 above)
Comment (xia-mc) [reply src/main/java/keystrokesmod/module/impl/other/Panic.java:20]: it shouldn't be happen.. we didn't add/remove any object from this list.

**Comment by TejasLamba2006** [comment]
@coderabbitai resolve

**Comment by SleepingPig35** [comment]
> Just asking but did you add the panic command in the help command?

Yes

**Comment by SleepingPig35** [comment]
So is this commit going to be pushed or not?

**Pull request merged**